presto-testing/src/main/java/io/prestosql/testing/datatype/DataTypeTest.java

```diff
@@ -55,27 +72,66 @@ public void execute(QueryRunner prestoExecutor, Session session, DataSetup dataS
         List<Object> expectedResults = inputs.stream().map(Input::toPrestoQueryResult).collect(toList());
         try (TestTable testTable = dataSetup.setupTestTable(unmodifiableList(inputs))) {
             MaterializedResult materializedRows = prestoExecutor.execute(session, "SELECT * from " + testTable.getName());
-            assertEquals(materializedRows.getTypes(), expectedTypes);
-            List<Object> actualResults = getOnlyElement(materializedRows).getFields();
-            assertEquals(actualResults.size(), expectedResults.size(), "lists don't have the same size");
-            for (int i = 0; i < expectedResults.size(); i++) {
-                assertEquals(actualResults.get(i), expectedResults.get(i), "Element " + i);
+            checkResults(expectedTypes, expectedResults, materializedRows);
+            if (runSelectWithWhere) {
+                String prestoQuery = buildPrestoQueryWithWhereClauses(testTable);
+                MaterializedResult filteredRows = prestoExecutor.execute(session, prestoQuery);
+                checkResults(expectedTypes, expectedResults, filteredRows);
```

Comment:
When this fails, it would be good to have a way to distinguish this failure from `checkResults` above.
Poor man's wold be to extract the body of this `if` section to a new method